### PR TITLE
Added rspec-mode

### PR DESCRIPTION
**Pull request by dcluna**
Based on [pezra's rspec-mode](https://github.com/pezra/rspec-mode).

### CODE REVIEW

**Comment by robbyoconnor** [comment]
This is in the ruby layer -- why do we need this? Just put it in the ruby layer

**Comment by robbyoconnor** [comment]
Also remove the files that do absolutely nothing -- and remove code that does absolutely nothing...

**Comment by CestDiego** [comment]
he's got a point.

**Comment by robbyoconnor** [comment]
This still hasn't been addressed...@dcluna :P

**Comment by dcluna** [comment]
I only added this layer because ruby-test-mode's keybindings weren't working for me, so I

**Comment by robbyoconnor** [comment]
It's good :)

File: layers/+lang/ruby/packages.el [hunk 1]
```diff
@@ -116,6 +117,23 @@
...
+      (evil-leader/set-key-for-mode 'enh-ruby-mode
+        "mps" 'rspec-verify-single
+        "mpa" 'rspec-verify-all
+        "mpr" 'rspec-rerun
+        "mpb" 'rspec-verify-matching
+        "mpc" 'rspec-verify-continue
+        "mpm" 'rspec-verify-method
+        "mpt" 'rspec-toggle-example-pendingness
+        ))))
+
```
Comment (robbyoconnor) [inline layers/+lang/ruby/packages.el]: Why the `mp` prefix?

**Comment by robbyoconnor** [comment]
In addition to this ***ALWAYS ADD A README WITH THE KEYBINDINGS! ****

File: layers/+lang/ruby/packages.el (hunk 1 above)
Comment (robbyoconnor) [inline layers/+lang/ruby/packages.el]: Why the `mp` prefix?  Shouldn't this be under the `mt` prefix per the [conventions](https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.org#tests)

**Comment by robbyoconnor** [comment]
In addition to this ***ALWAYS ADD A README WITH THE KEYBINDINGS!***

**Comment by robbyoconnor** [comment]
@dcluna 
> I only added this layer because ruby-test-mode's keybindings weren't working for me

Why not try to fix it? Perhaps you didn't have them installed properly...or the ruby you were using didn't have the gem available. I suspect this is the case. In addition, see the line comment.

**Comment by robbyoconnor** [comment]
Also, rename this as it is currently inaccurately titled ;)

**Comment by robbyoconnor** [comment]
Rebase again ;)

**Comment by StreakyCobra** [comment]
Now that  #3761 has been merged, this one is probably useless and can be closed :relaxed:

**Comment by syl20bnr** [comment]
Thank you !
I adapted this PR since #3761 has been merged, I added the key bindings of this PR in commit https://github.com/syl20bnr/spacemacs/commit/e9684f3
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**